lib/indentation-status-view.coffee

```diff
@@ -0,0 +1,113 @@
+    @appendChild(@tabLengthLink)
+    @handleEvents()
+    this
+
+  handleEvents: ->
+    @activePaneSubscription = atom.workspace.onDidChangeActivePaneItem =>
+      if @getActiveTextEditor()
+        @show()
+      @subscribeToPotentialEvents()
+
+    @tabTypeLink.addEventListener('click', @tabTypeToggle.bind(this))
+    @tabLengthLink.addEventListener('click', @tabWidthToggle.bind(this))
+
+    @tabTypeLinkClickSubscription = new Disposable => @tabTypeLink.removeEventListener('click', @tabTypeToggle)
+    @tabLengthLinkClickSubscription = new Disposable => @tabLengthLink.removeEventListener('click', @tabWidthToggle)
+
+    @subscribeToPotentialEvents()
+
+  hide: ->
+    @style.display = 'none'
+
+  show: ->
+    @style.display = ''
+
+  destroy: ->
+    @activePaneSubscription?.dispose()
+    @destroyPaneSubscription?.dispose()
+    @userChangeSubscription?.dispose()
+    @tabTypeLinkClickSubscription?.dispose()
+    @tabLengthLinkClickSubscription?.dispose()
+    @configSubscription?.dispose()
+    @indentationConfigSubscription?.dispose()
+
+  getActiveTextEditor: ->
+    atom.workspace.getActiveTextEditor()
+
+  tabTypeToggle: (ev) ->
+    editor = @getActiveTextEditor()
+    # Get current value
+    softTabs = editor.getSoftTabs(editor.getRootScopeDescriptor())
+    # Toggle softTabs
+    editor.setSoftTabs(not softTabs, editor.getRootScopeDescriptor())
+    # console.log 'toggle softTabs to ' + (not softTabs)
+    # console.log 'now it is ' + editor.getSoftTabs()
+
+    # Update interface
+    @updateTabTypeText()
+
+  tabWidthToggle: (ev) ->
+    editor = @getActiveTextEditor()
+
+    tabWidth = editor.getTabLength(scope: editor.getRootScopeDescriptor())
+
+    # Iterate over fixed options
+    tabWidth += 2
+    if tabWidth > 8
+      tabWidth = 2
+
+    editor.setTabLength(tabWidth)
+    @updateTabWidthText()
+
+  subscribeToPotentialEvents: ->
+    @userChangeSubscription?.dispose()
+    @indentationConfigSubscription?.dispose()
+    @indentationConfigSubscription = atom.config.observe 'editor.tabType', =>
+      @updateTabTypeText()
+    # FIXME: Noisy channel subscription, change to better one
+    @userChangeSubscription = @getActiveTextEditor()?.onDidChange =>
+      @updateIndentationText()
+    @updateIndentationText()
+
+  updateIndentationText: ->
+    @updateTabTypeText()
+    @updateTabWidthText()
+
+  updateTabTypeText: ->
+    editor = @getActiveTextEditor()
+    if editor
+      softTabs = editor.getSoftTabs()
+      if softTabs
+        @tabTypeLink.textContent = 'Space'
+      else
+        @tabTypeLink.textContent = 'Tab'
+    else
+      @hide()
+
+  updateTabWidthText: ->
+    editor = @getActiveTextEditor()
+    if editor
+      tabwidth = editor.getTabLength()
+    else
+      @hide()
+    # console.log 'update to ' + tabwidth
+    @tabLengthLink.textContent = '[ ' + tabwidth + ' ]'
+
+module.exports = document.registerElement('indentation-status', prototype: IndentationStatusView.prototype)
```

Comment:
We're trying to move away from custom elements. See https://github.com/atom/atom/issues/5503#issuecomment-188446070.